### DISCUSSION TITLE
JWKS auth support

### DISCUSSION THREAD

**Discussion by ivan-tkatchev** [dani-garcia/vaultwarden#5094] [Ideas]
Corporate web app proxies (like Teleport and possibly others) use JWKS to signal that a user has already been pre-authenticated by an upstream proxy.

https://auth0.com/docs/secure/tokens/json-web-tokens/json-web-key-sets

https://goteleport.com/docs/enroll-resources/application-access/jwt/introduction/

This is a well-known standard and allows seamless integration into on-prem authentication infrastructure. The user authenticates once at a proxy, and all webapps use the JWKS key to know that the user has already been athenticated for automatic log in without a password form. 

Any chance for Vaultwarden supporting this?

**Comment by BlackDex** [comment]
No. The Bitwarden clients (including the web-vault) need the actual plain text password to generate the correct hash and decrypt the users Private Key. Therefor this is not possible.

The only way for fully SSO Login is by having SSO Implemented and support for the Bitwarden Key Connector. The latter provides certificates/keys which allow the clients to decrypt the vault without the users password.

**Comment by ivan-tkatchev** [comment]
I understand. Is the planned SSO support tied heavily to OAuth?
This JWKS scheme is also an SSO scheme, the JWKS files are an RSA public key. Would that be enough?
